File: Npc/11003254.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11003254: Kaitlyn
/// </summary>
public class _11003254 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0403155707008175$
    // - What is it?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0403155707008176$
                // - I don't know what you did to impress $npcName:11003250[gender:0]$, but I'm not convinced. So, how are you going to prove that you're worth my time? 
                return 30;
            case (30, 1):
                // $script:0403155707008177$
                // - <font color="#909090">($npc:11003254[gender:1]$'s voice drops to a whisper.)</font>
                //   <font size='20'>This is all for Professor $npc:11003148[gender:0]$...</font>
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Next,
            (30, 1) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11003254.py
""" 11003254: Kaitlyn """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0403155707008175$
        # - What is it?
        return None # TODO

    def __30(self, pick: int) -> int:
        if self.index == 0:
            # $script:0403155707008176$
            # - I don't know what you did to impress $npcName:11003250[gender:0]$, but I'm not convinced. So, how are you going to prove that you're worth my time? 
            return 30
        elif self.index == 1:
            # $script:0403155707008177$
            # - <font color="#909090">($npc:11003254[gender:1]$'s voice drops to a whisper.)</font>
            #   <font size='20'>This is all for Professor $npc:11003148[gender:0]$...</font>
            return -1
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.NEXT
        elif (self.state, self.index) == (30, 1):
            return Option.CLOSE
        return Option.NONE
